### DISCUSSION TITLE
V12 IRequestPostProcessor, how to write with void return type IRequest?

### DISCUSSION THREAD

**Discussion by jafin** [LuckyPennySoftware/MediatR#847] [Q&A] [answered]
In V12 and the removal of `Unit` return type for a void IRequest, 
should an `IRequestPostProcessor` still be written with `Unit` for TResponse?  ie. `IRequestPostProcessor<VoidRequest, Unit>`

where VoidRequest is
```c#
public class VoidRequest: IRequest {}
```

But modify VoidRequest to be
```c#
public class VoidRequest: IRequest<Unit> {
   ...
   return Unit.Value;
}
```

Thanks,
Jason

**Comment by jbogard** [comment, marked as answer]
I added some notes to the upgrade guide around void requests and behaviors:

https://github.com/jbogard/MediatR/wiki/Migration-Guide-11.x-to-12.0

Short answer - leave your request alone, but you'll need to make your pipelines/processors return `Unit`. This was to avoid dual pipelines/processors etc:

```csharp
public class VoidRequest: IRequest {}
public class VoidRequestPostProcessor : IRequestPostProcessor<VoidRequest, Unit> {

    public Task Process(VoidRequest request, Unit response, CancellationToken cancellationToken)
    {
        // do some work, whatever
        return Task.CompletedTask;
    }
}
```

This works because I also removed the generic constraints on behaviors of `where TRequest : IRequest<TResponse>`.

**Reply by jafin** [reply]
Appreciate the response, that clarified things for me.
